src/utils/convert.js

```diff
@@ -825,278 +194,257 @@ const renderDefaultInputData = definition => {
-  templateContext.CODE = templateContext.CODE.replace(
-    "'use strict';\n",
-    ''
-  ).replace('"use strict";\n', '');
+const writeAuth = async (appDefinition, newAppDir) => {
+  const content = await renderAuth(appDefinition, appDefinition);
+  await createFile(content, 'authentication.js', newAppDir);
+};
 
-  const templateFile = path.join(TEMPLATE_DIR, '/scripting.template.js');
-  return renderTemplate(templateFile, templateContext, false);
+const writePackageJson = async (legacyApp, appDefinition, newAppDir) => {
+  const content = await renderPackageJson(legacyApp, appDefinition);
+  await createFile(content, 'package.json', newAppDir);
 };
 
-const writeScripting = (legacyApp, newAppDir) => {
-  return renderScripting(legacyApp).then(content => {
-    if (content) {
-      return createFile(content, 'scripting.js', newAppDir);
-    }
+const writeHydrators = async (appDefinition, newAppDir) => {
+  const content = await renderHydrators(appDefinition);
+  await createFile(content, 'hydrators.js', newAppDir);
+};
 
-    return null;
-  });
+const writeScripting = async (appDefinition, newAppDir) => {
+  await createFile(
+    appDefinition.legacy.scriptingSource,
+    'scripting.js',
+    newAppDir
+  );
 };
 
-const renderEnvironment = definition => {
-  const authFields = getAuthFieldKeys(definition);
-  const lines = _.map(authFields, key => {
-    const upperKey = _.snakeCase(key).toUpperCase();
-    return `${upperKey}=YOUR_${upperKey}`;
-  });
-  return lines.join('\n');
+const writeIndex = async (appDefinition, newAppDir) => {
+  const content = await renderIndex(appDefinition);
+  await createFile(content, 'index.js', newAppDir);
 };
 
-const writeEnvironment = (legacyApp, newAppDir) => {
-  const content = renderEnvironment(legacyApp);
-  if (!content) {
-    return Promise.resolve(null);
-  }
-  return createFile(content, '.env', newAppDir);
+const writeEnvironment = async (appDefinition, newAppDir) => {
+  const content = renderEnvironment(appDefinition);
+  await createFile(content, '.env', newAppDir);
 };
 
-const writeGitIgnore = newAppDir => {
+const writeGitIgnore = async newAppDir => {
   const srcPath = path.join(TEMPLATE_DIR, '/gitignore');
   const destPath = path.join(newAppDir, '/.gitignore');
-  return copyFile(srcPath, destPath).then(() => {
-    startSpinner('Writing .gitignore');
-    endSpinner();
-  });
+  await copyFile(srcPath, destPath);
+  startSpinner('Writing .gitignore');
+  endSpinner();
 };
 
-const convertApp = (legacyApp, newAppDir) => {
+const writeZapierAppRc = async newAppDir => {
+  const content = JSON.stringify({
+    includeInBuild: ['scripting.js']
+  });
+  await createFile(content, '.zapierapprc', newAppDir);
+  startSpinner('Writing .zapierapprc');
+  endSpinner();
+};
+
+const convertApp = async (
+  legacyApp,
+  appDefinition,
+  newAppDir,
+  silent = false
+) => {
+  if (silent) {
+    startSpinner = endSpinner = () => null;
```

Comment:
I did some googling and i don't think this has any chance of messing up the import for other files (it's just redefining a local variable). 

That said, I think i'd rather build this functionality into the spinner itself rather than do this each place we may need to maybe not have the spinner actually go. Maybe in utils, instead of exporting the object itself we export a wrapper class?

```js
// wrapper pseudocode
class SpinnerWrapperClass {
  constructor(silent=false)

  startSpinner() {
    if (this.silent) {return}
    spinner.spin()
}

// usage
const {spinner} = require('./utils/display')

const s = new Spinner(silent)
s.startSpinner()
```

slightly more verbose, but then we keep all functionality in the same spot. Not a blocker, just a thought.

Reply:
Currently, the `silent` option is there only for this [test](https://github.com/zapier/zapier-platform-cli/blob/4830272f39329485fc6a00ecb26aa3f691e6308b/src/tests/utils/convert.js#L62) to avoid printing stuff in the console while running tests. It's true that the spinner utils could use some improvement but it's probably not worth doing only for a test.